### PR TITLE
[MDB] Mirroring unused var removal PR.

### PR DESCRIPTION
**Pull request by MistakeNot4892**
Adoping #2576 as Noelle has moved on.

### CODE REVIEW

File: code/modules/integrated_electronics/core/detailer.dm [hunk 1]
```diff
@@ -48,5 +50,28 @@
...
+	scanning_color = FALSE
+	if(istype(target, /obj/item/electronic_assembly))
+		var/obj/item/electronic_assembly/target_assembly = target
+		detail_color = target_assembly.detail_color
+	if(istype(target, /obj/item/card/data))
+		var/obj/item/card/data/target_card = target
+		detail_color = target_card.detail_color
+	if(istype(target, /obj/item/integrated_electronics/detailer)) // why you'd want to copy off of another detailer, i wouldn't know
+		var/obj/item/integrated_electronics/detailer/target_detailer = target
+		detail_color = target_detailer.detail_color
```
Comment (afterthought2) [inline code/modules/integrated_electronics/core/detailer.dm]: I still hate this.

**Review by afterthought2 — DISMISSED** [review]

File: code/modules/integrated_electronics/core/detailer.dm (hunk 1 above)
Comment (MistakeNot4892) [reply code/modules/integrated_electronics/core/detailer.dm]: Think I dropped a commit somewhere because I fixed this once already.

Comment (MistakeNot4892) [reply code/modules/integrated_electronics/core/detailer.dm]: Pushed.

**Review by afterthought2 — APPROVED** [review]

File: code/modules/hydroponics/trays/tray_reagents.dm [hunk 2]
```diff
@@ -53,28 +53,28 @@
 // Weedkiller defines for hydroponics
 // *************************************
 
-//TODO: Remove those, they're completely unused
-/obj/item/weedkiller
+/obj/item/plantspray/weedkiller
```
Comment (PsyCommando) [inline code/modules/hydroponics/trays/tray_reagents.dm:56]: Why make a new weedkiller when there's another one above?

**Pull request merged**